src/plugins/WikiRouterPlugin/MediaWikiRouter.ts

```diff
@@ -0,0 +1,44 @@
+import WikiRouter from './WikiRouter';
+import { MwUtilGetUrl } from '@/@types/mediawiki';
+import {
+	inject,
+	InjectionKey,
+} from 'vue';
+
+export default class MediaWikiRouter implements WikiRouter {
+
+	private readonly getUrl: MwUtilGetUrl;
+
+	/**
+	 * The plugin uses a utilty to generate wiki urls from a wiki
+	 * page's title and ensure urls are consistent with various
+	 * mediawiki instances without relying on config variables or
+	 * constructing the url within the class itself.
+	 *
+	 * @see https://doc.wikimedia.org/mediawiki-core/master/js/#!/api/mw.util-method-getUrl
+	 *
+	 * @param {MwUtilGetUrl} getUrl
+	 */
+	public constructor(
+		getUrl: MwUtilGetUrl,
+	) {
+		this.getUrl = getUrl;
+	}
+
+	public goToTitle( title: string ): void {
+		window.location.href = this.getPageUrl( title );
+	}
+
+	private getPageUrl( title: string ): string {
+		return this.getUrl( title );
+	}
+
+}
+
+export const WikiRouterKey: InjectionKey<WikiRouter> = Symbol( 'WikiRouter' );
+
+export function useWikiRouter(): WikiRouter {
+	return inject( WikiRouterKey, () => {
+		throw new Error( 'No WikiRouter provided!' );
+	}, true );
+}
```

Comment:
Having this in `MediaWikiRouter.ts` feels strange to me… I think it would fit better in `WikiRouter.ts`, WDYT?

Reply:
yeah, that makes more sense. I'll move it!